prov/efa/src/rdm/efa_rdm_pke_cmd.c

```diff
@@ -573,16 +573,11 @@ void efa_rdm_pke_handle_send_completion(struct efa_rdm_pke *pkt_entry)
 		return;
 	}
 
-	/* These pkts are eager pkts withour hdrs */
-	if (pkt_entry->flags & EFA_RDM_PKE_SEND_TO_USER_RECV_QP) {
-		efa_rdm_pke_handle_eager_rtm_send_completion(pkt_entry);
-		efa_rdm_ep_record_tx_op_completed(ep, pkt_entry);
-		efa_rdm_pke_release_tx(pkt_entry);
-		return;
-	}
-
 	/* Start handling pkts with hdrs */
-	switch (efa_rdm_pke_get_base_hdr(pkt_entry)->type) {
+	switch (efa_rdm_pkt_type_of(pkt_entry)) {
+	case EFA_RDM_HEADERLESS_PKT:
```

Comment:
The comment in L576 is `/* Start handling pkts with hdrs */` :) 

It may be better to recover L 576-583 and drop the `case EFA_RDM_HEADERLESS_PKT` ?
It is more intuitive to distinguish the user recv qp send pkts and the local read pkt by flags explicitly, IMO